src/jdk.management.agent/share/classes/jdk/internal/agent/Agent.java

```diff
@@ -499,12 +499,17 @@ private static void startDiscoveryService(Properties props)
             }
 
             // Get service URL to broadcast it
-            Map<String,String> remoteProps = ConnectorAddressLink.importRemoteFrom(0);
-            String jmxUrlStr = remoteProps.get("sun.management.JMXConnectorServer.0.remoteAddress");
-
-            String instanceName = props.getProperty("com.sun.management.jdp.name");
-
-            JdpController.startDiscoveryService(address, port, instanceName, jmxUrlStr);
+            Map<String,String> remoteProps = null;
+            try {
+                remoteProps = ConnectorAddressLink.importRemoteFrom(0);
+            } catch (IOException ioe) {
+                warning(AGENT_EXCEPTION, "JDP not starting, PerfData not available: " + ioe.getMessage());
```

Comment:
Is this the only possible reason?

As per the JBS link I added there is a general problem about the Java side of the "Perf" code not knowing/realizing that `UsePerfData` can be disabled. Ideally there would be a way to query this, or a specific type of exception that could be thrown to report it.

Will the user understand what this message means? Should it explicitly say "please ensure -XX:-UsePerfData" is not set?

Reply:
> Is this the only possible reason?

Yes I think it is - although IOException is already declared by importRemoteFrom, and that calls perf.attach(id) which declares IOException, the native Perf_Attach doesn't look like it throws it (OOM and IllegalArgument can be thrown).

BUT maybe a better alternative is to catch the BufferUnderflowException, this is the unchecked Exception which currently terminates us.  This is more specific and safer in case we ever do make more use of IOException in the attach - I wil update...

(Use of ConnectorAddressLink is all about communication within the same process so it doesn't expect any problems...)


> Will the user understand what this message means?

If I use the BufferUnderflowException, there's no detail message so it will say:

Warning: Exception thrown by the agent : JDP not starting, PerfData not available

If you have added -UsePerfData you should see the connection, it's not too chatty, and isn't saying too much in case it's wrong and the perfdata failed to allocate for some other reason.